tests/Makefile

```diff
@@ -21,13 +21,19 @@ BIN_DIR := $(ROOT_DIR)/build/bin
 #TOP_DIR := $(ROOT_DIR)
 ECHOE = /bin/echo -e
 
-SRC_DIRS = $(ROOT_DIR)/src
-INC_DIRS := $(ROOT_DIR)/include
-INC_DIRS += $(ROOT_DIR)/../include
-
 # Switch on Linux & disable asserts for the testing suites
 CFLAGS += -DNDEBUG
+ifneq ($(DGTEST),1)
```

Comment:
I assume this is your `VARIANT`

Reply:
There's a testing suite for C & there must be a testing suite for C++, so you need two makefiles, because the tests are completely different.

you can't run the C tests under the C++ Testing suite, you need a C++ testing suite to test the functionality is working being called from C++?

Reply:
I am managing this with same makefile.